### PR TITLE
Add support for Flatpak automatic generation

### PR DESCRIPTION
**Pull request by skadge**
Flatpak (https://flatpak.org/) is becoming the dominant application distribution mechanism on Linuxes.

This PR depends on (and indeed, includes) PR #19 and #21 .

This PR:
- add supports for building FlatFab as a flatpak
- add a github workflow action to automatically build the flatpak when a tag is pushed

image

Example result (on fake tags I created for testing):

image

TODO:
- [x] the resulting flatpak does not currently work due to a dynamic linking issue (`libGLU.so.1` inside of the flatbak container is not found, unclear why, a workaround would be to start FlatFab from a script that add `/app/lib64` to `LD_LIBRARY_PATH`)
- [x] we should probably only run this workflow for tagged releases

Optional:

- ideally, we should automatically publish the flatpak on flathub (https://flathub.org/) which is the leading 'app store' for open-source flatpaks. I haven't looked into it yet.

@JamesMcCrae please let me know if you think this is a useful direction, and if you feel like providing this kind of distribution channel (this would definitely make it much easier to install and update FlatFab on linux, without having to care too much about the different linux distributions)

### CODE REVIEW

**Comment by skadge** [comment]
For reference, the generate binary is only ~1.5MB big! (this is possible as all the Qt libraries are manage by flatpak as shared runtimes, thus massively reducing the size of the package)

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 set(CMAKE_AUTOUIC ON)
 
-find_package(Qt5 COMPONENTS Core Gui OpenGL Widgets Network WebEngineWidgets REQUIRED)
+find_package(Qt5 COMPONENTS Core Gui OpenGL Widgets Network REQUIRED)
```
Comment (JamesMcCrae) [inline CMakeLists.txt]: Can "Network" also be removed here?

File: CMakeLists.txt [hunk 2]
```diff
@@ -78,10 +78,16 @@ target_link_libraries(${PROJECT_NAME}
     Qt5::OpenGL
     Qt5::Widgets
     Qt5::Network
```
Comment (JamesMcCrae) [inline CMakeLists.txt]: Remove "Qt5::Network" here?

File: README.md [hunk 3]
```diff
@@ -21,3 +21,22 @@ You need Qt5 (tested on Ubuntu 24.10 with QtCreator 13; any previous release sho
...
+#### Flatpak
+
+Uses the latest Qt 5.15 runtime; builds manually Eigen and GLU.
+
+To build:
+
+- install `flatpak` and `flatpak-builder`
+- `flatpak install org.kde.Sdk/x86_64/5.15-24.08`
+- `flatpak install org.kde.Platform/x86_64/5.15-24.08`
+- `flatpak install io.qt.qtwebengine.BaseApp/x86_64/5.15-24.08`
```
Comment (JamesMcCrae) [inline README.md]: Should this line be removed, as it appears specific to webengine?

**Review by JamesMcCrae — CHANGES_REQUESTED** [review]
This seems like a great initiative, as it will streamline distribution of latest releases on Linux moving forward.  A few comments just relating to further removal of Network/Webengine from these automation files, and then happy to approve the changes.

**Comment by JamesMcCrae** [comment]
Re: GLU library issue, where does the Flatpak-generated build look for the library?  When I run "ldd FlatFab" locally I get:

	libGLU.so.1 => /lib/x86_64-linux-gnu/libGLU.so.1 (0x00007f5df29ef000)

File: CMakeLists.txt (hunk 1 above)
Comment (skadge) [reply CMakeLists.txt]: good catch. Fixed in #21

File: CMakeLists.txt (hunk 2 above)
Comment (skadge) [reply CMakeLists.txt]: same. Fixed in #21

File: README.md (hunk 3 above)
Comment (skadge) [reply README.md]: fixed, thanks

**Comment by skadge** [comment]
> Re: GLU library issue, where does the Flatpak-generated build look for the library? When I run "ldd FlatFab" locally I get:
> 
> ```
> libGLU.so.1 => /lib/x86_64-linux-gnu/libGLU.so.1 (0x00007f5df29ef000)
> ```

As you might have seen in the flatpak manifest, I build manually libGLU when building the flatpak. The library end up in `/app/lib64`. For some reason (and I could not figure out why), the FlatFab binary happily compiles against this library, but  fails to launch, as `/app/lib64` is not in the `LD_LIBRARY_PATH` of the built flatpak. So in `flatpak/flatfab-launch.sh`, I manually append this path the the `LD_LIBRARY_PATH` before launching FlatFab.

**Comment by skadge** [comment]
@JamesMcCrae : I've renamed the flatpak id from `com.flatfab.FlatFab` to `edu.toronto.dgp.FlatFab`.

**Review by JamesMcCrae — APPROVED** [review]

**Pull request merged**